### ISSUE TITLE
Querying: Slow 'GetSearchResult' request

### ISSUE THREAD

**Issue by SRandomName** [skttl/umbraco-fulltextsearch8#64]
Hello!

We've implemented FullTextSearch V1.2.1-beta in our project.
We're using Umbraco V8.14.1 with Examine V1.2.2.

Is it normal that the 'GetSearchResult' request takes +- 12 seconds to complete using the backoffice UI of fullTextSearch?
![SlowFullTextSearch](https://user-images.githubusercontent.com/48089189/133389041-26d9aa0d-d29c-4682-8a5c-c74f9657f485.gif)
Thanks!

**Comment by skttl** [comment]
No, that is not normal. How long does it take for the same search in the examine dashboard?

**Comment by skttl** [comment]
And can you show your configuration?

**Comment by SRandomName** [comment]
The same search in the examine dashboard takes 100ms. 
And this is my configuration of fulltextsearch: 
![image](https://user-images.githubusercontent.com/48089189/133390638-a2136544-6116-4943-8bf0-c642b825ea99.png)
![image](https://user-images.githubusercontent.com/48089189/133390671-5991d736-56ad-4ca5-96f7-4f9c441f5a63.png)

**Comment by SRandomName** [comment]
@skttl Is there anything we can do to fix this?  :-)

**Comment by skttl** [comment]
I'm not sure why it does that. Is it the same on the frontend?

**Comment by skttl** [comment]
If you set your logging level to debug, it should log the query used for searching with the template `Trying to search for {query}`. Try doing that, and see what the query ends up with, then try searching the examine dashboard with the same query and see if there is a difference.

**Comment by SRandomName** [comment]
Ok will do, i've also tried to uninstall and re-install. Weird thing was that I had V1.2.1-beta installed while the latest nuget version for Umbraco 8 is V1.2.0

**Comment by SRandomName** [comment]
Full text search query: 
```Trying to search for "( ((nodeName:beton~0.8^10 OR nodeName_nl:beton~0.8^10) (FullTextContent:beton~0.8 OR FullTextContent_nl:beton~0.8) ) ) AND (__IndexType:content AND ((__VariesByCulture:y AND __Published_nl:y) OR (__VariesByCulture:n AND __Published:y))) AND -(__NodeTypeAlias:forbidden forgotPasswordMember formSucces login pageNotFound resetPasswordMember search testsOffer testItem) AND -(templateID:0)"```

The result are the same: 
![image](https://user-images.githubusercontent.com/48089189/133409194-6f0af532-38a5-4506-8692-18d92efc44f6.png)

**Comment by SRandomName** [comment]
@skttl  when debuggin i'm getting this error: 
![image](https://user-images.githubusercontent.com/48089189/133410752-b47a6079-8c58-4047-ba95-abed880371d7.png)

**Comment by skttl** [comment]
What about the speed of search in the examine dashboard? Is that also slow with that query?

**Comment by SRandomName** [comment]
It takes ~100ms

**Comment by SRandomName** [comment]
Maybe important: Our lucene version is v3.0.3

**Comment by skttl** [comment]
Could be related to #54 maybe, and then in the controller for the dashboard it runs `.Select()` on the results.
https://github.com/skttl/umbraco-fulltextsearch8/blob/develop/src/Our.Umbraco.FullTextSearch/Controllers/IndexController.cs#L269

**Comment by SRandomName** [comment]
The requests are also taking a lot of time when using the searchService in code.

**Comment by SRandomName** [comment]
@skttl  I found the problem. We have a large <svg> in our master template. Excluding svg in xpath fixes the problem.
I was reading the documentation and I was wondering how to use the 'IsIndexingActive' to exclude content from being rendered.

**Comment by SRandomName** [comment]
@skttl  - I found how to use the IsIndexingActive. Now when re-indexing our pages there is another problem .. :-/ 
In OurUmbracoFullTextRendering you render the page by using the @Umbraco.RenderTemplate(Model.Id). 

We made a custom controller and viewmodel for each document type. Controller code for example always starts with: 
`var vm = new ToolViewModel(_umbraco.AssignedContentItem);`

where _umbraco is the UmbracoHelper.

Problem now is that the IpublishedContent custom properties (textstring, datetime, ...) are always null.
Any idea why?

**Comment by skttl** [comment]
So did it go faster in the dashboard too, when you got the svgs out of the index? How big are those svgs?

**Comment by SRandomName** [comment]
Yes, it was way faster in the dasboard too when we got the svgs out of the index. 
It was a very large svg with several icons (600+ lines)

**Comment by SRandomName** [comment]
Hello @skttl  - An update on this. We've implemented this package into another simple project without any svg's or anything. Using the Fulltextsearch UI calls take +- 600ms which is too long. Same calls in examine ui take +-100ms. Any idea what the problem could be?

**Comment by skttl** [comment]
Could be some innefficiencies in the summarizer. How much text is in the "FullTextContent" field on the largest page you have?

**Comment by SRandomName** [comment]
There is one page with +- 10 000 characters in the full text content field.

**Comment by SRandomName** [comment]
@skttl It seems like adding a lot of disallowed aliases > content types slows down the querying. Is that possible?

**Comment by skttl** [comment]
Could be what's in your config? -------- Oprindelig besked --------Fra: RandomName123 ***@***.***>Dato: ons. d. 6. okt. 2021 16.03Til: skttl/umbraco-fulltextsearch8 ***@***.***>Cc: Søren Kottal ***@***.***>, Mention ***@***.***>Emne: Re: [skttl/umbraco-fulltextsearch8] Querying: Slow 'GetSearchResult' request (#64)
@skttl It seems like adding a lot of disallowed aliases > content types slows down the querying. Is that possible?
